src/Core/Utilities/CoreHelpers.cs

```diff
@@ -555,12 +555,13 @@ private static string Other2Qwerty(string value, string otherMap, string qwertyM
             return sb.ToString();
         }
 
-        public static string SanitizeForEmail(string value)
+        public static string SanitizeForEmail(string value, bool htmlEncode = true)
         {
             var cleanedValue = value.Replace("@", "[at]")
+                .Replace(".", "[dot]")
```

Comment:
One suggestion, because you will have legit org names that look like this: `Google Pvt. Ltd.` and therefore only want to flag/tokenize "dangerous dots" which are not followed by whitespace (indicating a domain name could be formed).

Here you could either use the `\w` word character match after dot, OR, the `\S` non white-space character match, not sure which is more appropriate but in testing both worked for most scenarios when I checked them on RegexStorm.

Also, this removes any ability to double-inject `https` into a pattern so when we remove the first we retain the 2nd, making the replacement of these recursive.

```c#
var cleanedValue = value.Replace("@", "[at]");
var regexOptions = RegexOptions.CultureInvariant |
    RegexOptions.Singleline |
    RegexOptions.IgnoreCase;
cleanedValue = Regex.Replace(cleanedValue, @"(\.\w)",
        m => string.Concat("[dot]", m.ToString().Last()), regexOptions);
while (Regex.IsMatch(cleanedValue, @"((^|\b)(\w*)://)", regexOptions))
{
    cleanedValue = Regex.Replace(cleanedValue, @"((^|\b)(\w*)://)",
        string.Empty, regexOptions);
}
```

Reply:
Implemented as suggested & added a screenshot showing the functioning condition for periods.